pkg/accesscontrol/policy_rule_index.go

```diff
@@ -85,25 +87,29 @@ func (p *policyRuleIndex) get(subjectName string) *AccessSet {
 
 func (p *policyRuleIndex) addAccess(accessSet *AccessSet, namespace string, roleRef rbacv1.RoleRef) {
 	for _, rule := range p.getRules(namespace, roleRef) {
-		for _, group := range rule.APIGroups {
-			for _, resource := range rule.Resources {
-				names := rule.ResourceNames
-				if len(names) == 0 {
-					names = []string{All}
-				}
-				for _, resourceName := range names {
-					for _, verb := range rule.Verbs {
-						accessSet.Add(verb,
-							schema.GroupResource{
-								Group:    group,
-								Resource: resource,
-							}, Access{
-								Namespace:    namespace,
-								ResourceName: resourceName,
-							})
+		if len(rule.Resources) > 0 {
+			for _, group := range rule.APIGroups {
+				for _, resource := range rule.Resources {
+					names := rule.ResourceNames
+					if len(names) == 0 {
+						names = []string{All}
+					}
+					for _, resourceName := range names {
+						for _, verb := range rule.Verbs {
+							accessSet.Add(verb,
+								schema.GroupResource{
+									Group:    group,
+									Resource: resource,
+								}, Access{
+									Namespace:    namespace,
+									ResourceName: resourceName,
+								})
+						}
 					}
 				}
 			}
+		} else if roleRef.Kind == clusterRoleKind {
```

Comment:
Should this check that the binding is a clusterrolebinding and not a rolebinding? Seems like for clusterrolebinding we're setting namespace as `All` (which is `*`)

Reply:
I'm checking if the kind is "ClusterRole" and not setting the namespace, does it mean "All" in k8s terms ?

Reply:
I have this test at GrantsNonResource, should I replicate it here ?